src/GUI/GUIEnums.h

```diff
@@ -0,0 +1,344 @@
+    MENU_MAIN = 0,
+    MENU_NEWGAME = 1,
+    MENU_CREDITS = 2,
+    MENU_SAVELOAD = 3,
+    MENU_EXIT_GAME = 4,
+    MENU_5 = 5,
+    MENU_CREATEPARTY = 6,
+    MENU_NAMEPANELESC = 7,
+    MENU_CREDITSPROC = 8,
+    MENU_LoadingProcInMainMenu = 9,
+    MENU_DebugBLVLevel = 10,
+    MENU_CREDITSCLOSE = 11,
+    MENU_MMT_MAIN_MENU = 12,
+    MENU_NONE = -1,
+};
+
+
+enum WindowType {
+    WINDOW_null = 0,
+    WINDOW_MainMenu = 1,
+    WINDOW_GameMenu = 3,
+    WINDOW_CharacterRecord = 4,
+    WINDOW_Options = 6,
+    WINDOW_8 = 8,
+    WINDOW_Book = 9,
+    WINDOW_Dialogue = 10,
+    WINDOW_QuickReference = 12,
+    WINDOW_F = 15,
+    WINDOW_Rest = 16,
+    WINDOW_ChangeLocation = 17,
+    WINDOW_SpellBook = 18,
+    WINDOW_GreetingNPC = 19,
+    WINDOW_Chest = 20,
+    WINDOW_22 = 0x16,
+    WINDOW_SaveLoadButtons = 23,
+    WINDOW_MainMenu_Load = 0x18,
+    WINDOW_HouseInterior = 0x19,
+    WINDOW_Transition = 26,
+    WINDOW_CastSpell = 27,  // OnCastTargetedSpell
+    WINDOW_Scroll = 0x1E,
+    WINDOW_CastSpell_InInventory = 31,
+    WINDOW_GameOverWindow = 70,
+    WINDOW_50 = 80,  // Debug
+    WINDOW_59 = 89,  // Debug: Item Generation Window
+    WINDOW_PressedButton2 = 90,      // OnButtonClick2?
+    WINDOW_CharactersPressedButton = 91,
+    WINDOW_PressedButton = 92,
+    WINDOW_5D = 93,
+    WINDOW_SaveLoadBtn = 94,
+    WINDOW_LoadGame_CancelBtn = 95,  // OnCancel3?
+    WINDOW_CloseRestWindowBtn = 96,
+    WINDOW_ExitCharacterWindow = 97,  // OnCancel2
+    WINDOW_RestWindow = 0x62,
+    WINDOW_BooksButtonOverlay = 99,
+    WINDOW_CharacterWindow_Stats = 0x64,
+    WINDOW_CharacterWindow_Skills = 0x65,
+    WINDOW_CharacterWindow_Awards = 0x66,
+    WINDOW_CharacterWindow_Inventory = 0x67,
+    WINDOW_GameOptions = 104,
+    WINDOW_KeyMappingOptions = 0x69,
+    WINDOW_VideoOptions = 0x6A,
+    WINDOW_LloydsBeacon = 177,
+    WINDOW_TownPortal = 195,
+    WINDOW_QuestBook = 200,
+    WINDOW_AutonotesBook = 0xC9,
+    WINDOW_MapsBook = 0xCA,
+    WINDOW_CalendarBook = 0xCB,
+    WINDOW_JournalBook = 0xE0,
+    WINDOW_Unknown,               // i wasnt able to find real value for those
+    WINDOW_CharacterCreation,     // new addition, because i wasnt able to find real value for this
+    WINDOW_CharacterCreationBtn,  // new addition, because i wasnt able to find real value for this
+    WINDOW_GenericCancel,         // new addition, because i wasnt able to find real value for this
+    WINDOW_GameUI,                // new addition, because i wasnt able to find real value for this
+    WINDOW_Credits,               // new addition, because i wasnt able to find real value for this
+    WINDOW_Save,                  // new addition, because i wasnt able to find real value for this
+    WINDOW_Load,                  // new addition, because i wasnt able to find real value for this
+    WINDOW_DebugMenu,
+    // =======
+    // ToString (below) is updated until this point
+};
+
+MM_DECLARE_SERIALIZATION_FUNCTIONS(WindowType)
```

Comment:
Please also add GUIEnums.cpp and move the serialization implementation code there (`MM_DEFINE_ENUM_MAGIC_SERIALIZATION_FUNCTIONS` invocations).